modules/modelSetup/mixin/ModelSetupDiffusionNoiseMixin.py

```diff
@@ -55,15 +56,26 @@ def _get_timestep_discrete(
             generator: Generator,
             batch_size: int,
             args: TrainArgs,
+            global_step: int,
     ) -> Tensor:
         if not deterministic:
-            return torch.randint(
-                low=0,
-                high=int(noise_scheduler.config['num_train_timesteps'] * args.max_noising_strength),
-                size=(batch_size,),
-                generator=generator,
-                device=generator.device,
-            ).long()
+            min_timestep = int(noise_scheduler.config['num_train_timesteps'] * args.min_noising_strength)
+            max_timestep = int(noise_scheduler.config['num_train_timesteps'] * args.max_noising_strength)
+            if args.noising_weight == 0:
+                return torch.randint(
+                    low=min_timestep,
+                    high=max_timestep,
+                    size=(batch_size,),
+                    generator=generator,
+                    device=generator.device,
+                ).long()
+            else:
+                np.random.seed(global_step)
```

Comment:
I don't really like this line. It changes the global seed of the rng. This means calling this function has some unexpected side effects. Calling `np.random.default_rng(seed)`, than using the returned rng might be better.

This does have some small performance impact, because a new rng is created. But I don't think the returned rng supports changing the seed. Otherwise you could save a single instance and change the seed on every call to _get_timestep_discrete